### PR TITLE
Add images for CSI

### PR DESCRIPTION
**Pull request by rayandas**
#### Pull Request Checklist ####

- [x] Change does not remove any existing Images or Tags in the images-list file
- [x] Change does not remove / overwrite exiting Images or Tags in Rancher DockerHub
- [x] If updating an existing entry, verify the `SOURCE` is still accurate and upstream hasn't been migrated to a new regitry or repo (if they've migrated, a new repo request to EIO is needed to comply with the `SOURCE DESTINATION TAG` pattern)
- [x] New entries are in format `SOURCE DESTINATION TAG`
- [x] New entries are added to the correct section of the list (sorted lexicographically)
- [x] New entries have a repo created in Rancher Dockerhub (where the image will be mirrored to)
- [ ] Changes to scripting or CI config have been tested to the best of your ability

#### Types of Change ####
New images for the vSphere CSI 2.3.0 update.

#### Linked Issues ####
https://github.com/rancher/rancher/issues/33848

#### Final Checks after the PR is merged ####
- [ ] Confirm that you can pull the new images and tags from DockerHub

### CODE REVIEW

File: images-list
Comment (PennyScissors) [inline images-list:261]: This is a Longhorn image and shouldn't be used in the vSphere charts as it can be confusing for users regardless of the source image being essentially the same. So this image should actually be:
`k8s.gcr.io/sig-storage/csi-resizer rancher/mirrored-sig-storage-csi-resizer v1.3.0`

However, the latest image the chart uses is from Quay (see [here](https://github.com/rancher/charts/blob/b35ff06e9138b1f221f1b51465755af7ac0ba797/charts/rancher-vsphere-csi/rancher-vsphere-csi/100.1.0%2Bup2.3.0/values.yaml#L33), and [here](https://github.com/rancher/image-mirror/blob/6fb5954a66ba9fe4d3d9230efc4fc55706e86df2/images-list#L541)), so you will need to submit a request to EIO to get the image repository created before it can be mirrored.

File: images-list
Comment (PennyScissors) [inline images-list:549]: This is incorrect, the chart is using a [new image](https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/28c55a22dcb986460b66f7de3838c96146e39900/manifests/vanilla/vsphere-csi-driver.yaml#L307) from a different registry so a new repo will need to be created and the image will need a new name. This should be:
`k8s.gcr.io/sig-storage/livenessprobe:v2.4.0 rancher/mirrored-sig-storage-livenessprobe v2.4.0`

File: images-list
Comment (PennyScissors) [inline images-list:542]: Same as the  livenessprobe image, the registry changed. This should be:
`k8s.gcr.io/sig-storage/csi-node-driver-registrar rancher/mirrored-sig-storage-csi-node-driver-registrar v2.3.0`
New image: https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/28c55a22dcb986460b66f7de3838c96146e39900/manifests/vanilla/vsphere-csi-driver.yaml#L396

**Review by PennyScissors — CHANGES_REQUESTED** [review]
Looks like [this image](https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/28c55a22dcb986460b66f7de3838c96146e39900/manifests/vanilla/vsphere-csi-driver.yaml#L345) is missing from this PR. Same situation as the livenessprobe image, it needs a new repo, name, and should be added here as:
`k8s.gcr.io/sig-storage/csi-provisioner rancher/mirrored-sig-storage-csi-provisioner v3.0.0`

File: images-list
Comment (rayandas) [reply images-list:261]: Sure I will update the PR with these changes.

File: images-list
Comment (rayandas) [reply images-list]: Updated the PR with all the changes. it can be merged when EIO issue will be closed.

File: images-list
Comment (PennyScissors) [inline images-list:267]: Was a repo for this image created? I don't see it in the EIO issue.

Comment (rayandas) [reply images-list:267]: Yes, here's the issue number is #730

**Review by PennyScissors — APPROVED** [review]

**Comment by rayandas** [comment]
I don't have merge access. Can anyone of you please merge it? @PennyScissors @kinarashah

**Review by kinarashah — APPROVED** [review]

**Pull request merged**
